src/utils.zig

```diff
@@ -1415,6 +1415,27 @@ pub const BinaryReader = struct {
 		return .{.remaining = data, .endian = endian};
 	}
 
+	pub fn readVec(self: *BinaryReader, T: type) error{OutOfBounds, IntOutOfBounds}!T {
+		switch(@typeInfo(T)) {
+			.vector => |typeInfo| {
+				var result: T = undefined;
+				for(0..typeInfo.len) |i| {
```

Comment:
Should be an inline for. The compiler can generate better code if the element index of the vector is comptime known.